icu4j/main/classes/core/src/com/ibm/icu/impl/locale/XLikelySubtags.java

```diff
@@ -199,8 +199,8 @@ public LSR makeMaximizedLsrFrom(ULocale locale) {
 
     public LSR makeMaximizedLsrFrom(Locale locale) {
         String tag = locale.toLanguageTag();
-        if (tag.startsWith("x-")) {
-            // Private use language tag x-subtag-subtag...
+        if (tag.startsWith("und-x-")) {
```

Comment:
I don't know whether Java Locale will prepend "und-". Please check for both forms.